test/blackbox-tests/test-cases/build-bin.t

```diff
@@ -0,0 +1,22 @@
+Pform expansion should work in `exec` but also in `build`:
+
+  $ cat > dune-project << EOF
+  > (lang dune 3.10)
+  > (package (name e))
+  > EOF
+  $ cat > dune << EOF
+  > (executable (public_name e))
```

Comment:
You should also test the case where the binary isn't public but is used in made public using `(binaries ..)` in the env stanza.

Reply:
This reveals a rather interesting behavior, as `dune exec %{bin:private}` works without issues, but using it in `env` triggers an error that the target (not the source!) doesn't have a rule:

```
  File "dune", line 4, characters 30-34:
  4 |   (_ (binaries private.exe as priv)))
                                    ^^^^
  Error: No rule found for priv
```

Reply:
Nevermind, it was missing some parens so it got parsed incorrectly (which is problematic in some way but not relevant to this PR).